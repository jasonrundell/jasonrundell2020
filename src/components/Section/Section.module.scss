.root {
  width: 100%;
}

.contained {
  max-width: 42rem;
}

.centered {
  margin-left: auto;
  margin-right: auto;
}
