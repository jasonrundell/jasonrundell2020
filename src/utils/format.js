/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
