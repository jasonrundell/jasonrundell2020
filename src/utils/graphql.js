export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
