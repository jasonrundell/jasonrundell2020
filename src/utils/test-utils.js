export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
