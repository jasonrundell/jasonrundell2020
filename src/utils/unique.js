const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
