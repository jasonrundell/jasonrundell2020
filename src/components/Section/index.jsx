import React from 'react'
import PropTypes from 'prop-types'

import { classNames } from '../../utils/css'

// CSS
import { root } from './Section.module.scss'

const Section = ({ id, children, className }) => {
  const classes = classNames(root, className)

  return (
    <section id={id} className={classes}>
      {children}
    </section>
  )
}

Section.defaultProps = {
  id: null,
  className: null,
}

Section.propTypes = {
  id: PropTypes.string,
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
}

export default Section
