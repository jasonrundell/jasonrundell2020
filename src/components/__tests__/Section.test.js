import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Section from '../Section'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Section component', () => {
  it('Section renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <Section>Section component</Section>
          <Section id="thisistheid">Section with id</Section>
          <Section className="class__name--test">
            Section with className
          </Section>
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Section has required children prop', () => {
    act(() => {
      render(<Section>Section component</Section>, container)
    })
    expect(container.textContent).toBe('Section component')
  })

  it('Section id prop', () => {
    act(() => {
      render(
        <Section id="thisistheid">Section with id=thisistheid</Section>,
        container
      )
    })
    expect(container.textContent).toBe('Section with id=thisistheid')
  })

  it('Section className prop', () => {
    act(() => {
      render(
        <Section className="class__name--test">Section with className</Section>,
        container
      )
    })
    expect(container.textContent).toBe('Section with className')
  })
})
