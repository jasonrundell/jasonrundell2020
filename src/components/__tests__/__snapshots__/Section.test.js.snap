// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component Section renders correctly 1`] = `
<div>
  <section
    className="root"
    id={null}
  >
    Section component
  </section>
  <section
    className="root"
    id="thisistheid"
  >
    Section with id
  </section>
  <section
    className="root class__name--test"
    id={null}
  >
    Section with className
  </section>
</div>
`;
